torch/_inductor/ir.py

```diff
@@ -1494,6 +1496,16 @@ def codegen_reference(self):
         return f"{as_strided}({self.get_name()}, {size}, {stride})"
 
 
+class InplaceView(ReinterpretView):
+    def codegen_reference(self):
+        size = V.graph.sizevars.codegen_shape_tuple(self.layout.size)
+        stride = V.graph.sizevars.codegen_shape_tuple(self.layout.stride)
+        offset = V.graph.sizevars.codegen_sizevar(self.layout.offset)
+        if offset != "0":
+            return f"{self.get_name()}.as_strided_({size}, {stride}, {offset})"
+        return f"{self.get_name()}.as_strided_({size}, {stride})"
```

Comment:
I think this will only work if we just return the mutated tensor but never use it.  This is because the inductor metadata will be wrong.  We need to do something like:
```
assert isinstance(self, TensorBox)
self.data = InplaceView(self.data, ....)
return self
```

We have a mutable TensorBox.data to allow this type of pattern.  (While StorageBox.data allows data mutation.)

Please add a test where the mutated tensor is used somehow.

Reply:
We also may want to codegen an explit call to mutate the metadata in case it isn't used we never call codegen_reference().

Reply:
Thanks for your comments. I updated codes in commit d97ffd3. Does this change make sense?